src/Frontend/Page/Admin.hs

```diff
@@ -342,15 +402,36 @@ adminSettingsGaPUsersCreate =
 
 adminSettingsGaPClassesView :: ActionM r m => m (Frame PageAdminSettingsGaPClassesView)
 adminSettingsGaPClassesView =
-    makeFrame =<< PageAdminSettingsGaPClassesView . mapMaybe toClass <$> persistent getSpaces
-  where
-    toClass (ClassSpace clss) = Just clss
-    toClass _                 = Nothing
+    makeFrame =<< PageAdminSettingsGaPClassesView <$> persistent getSchoolClasses
 
 adminSettingsGaPClassesCreate :: ActionM r m => m (Frame PageAdminSettingsGaPClassesCreate)
 adminSettingsGaPClassesCreate =
     makeFrame PageAdminSettingsGaPClassesCreate
 
+adminSettingsGaPUserEdit :: ActionM r m => AUID User -> ServerT (FormHandler PageAdminSettingsGaPUsersEdit ST) m
+adminSettingsGaPUserEdit uid = redirectFormHandler editUserPage editUser
+  where
+    editUserPage = persistent $
+        PageAdminSettingsGaPUsersEdit
+        <$> ((\(Just u) -> u) <$> findUser uid) -- FIXME: Error handling
+        <*> getSchoolClasses
+
+    editUser (EditUser role clss) = do
+        let isSelectedClass (Student clss')    = clss == clss'
+            isSelectedClass (ClassGuest clss') = clss == clss'
+            isSelectedClass _                  = False
+
+        let addGroup gs = case role of
+                RoleStudent -> Student clss : gs
+                RoleGuest   -> ClassGuest clss : gs
+
+        persistent $ modifyUser uid (userGroups %~ addGroup . filter (not . isSelectedClass))
```

Comment:
We agreed that the userGroup list will contain classes from the past. The semantics is not well defined here. I would leave the actual implementation as it is now, and work on the full solution in a different PR. Of course I am going to write a documentation comment what is happening now. Maybe some tests, which are also convenient here.